### PR TITLE
fix(cosmic-swingset): apply Far/Data everywhere I could think of

### PR DESCRIPTION
**Pull request by warner**
@michaelfig I added Far/Data everywhere I could think of, but the cosmic-swingset tests are still failing (with `marshal.js`'s throw-on-unmarked-empty-objects asserts uncommented). I think you can probably spot the problem faster than me, also I think there's a lot of code that isn't covered by the tests that I'd probably miss. Can you move this forward?

Switch to this branch, then apply the following patch to marshal.js:

```diff
diff --git a/packages/marshal/src/marshal.js b/packages/marshal/src/marshal.js
index 1cb78971..6b58d3f8 100644
--- a/packages/marshal/src/marshal.js
+++ b/packages/marshal/src/marshal.js
@@ -293,8 +293,8 @@ function isPassByCopyRecord(val) {
   // error, then (TODO) it will go back to pass-by-copy.
   // See https://github.com/Agoric/agoric-sdk/issues/2018
   if (descKeys.length === 0 && proto === objectPrototype) {
-    // console.log(`--- @@marshal: empty object without Data/Far/Remotable`);
-    // assert.fail(X`empty object without Data/Far/Remotable`);
+    console.log(`--- @@marshal: empty object without Data/Far/Remotable`);
+    assert.fail(X`empty object without Data/Far/Remotable`);
     return false;
   }
   for (const descKey of descKeys) {
@@ -512,8 +512,9 @@ export function passStyleOf(val) {
         return 'copyRecord';
       }
       assertRemotable(val);
-      // console.log(`--- @@marshal: pass-by-ref object without Far/Remotable`);
-      // assert.fail(X`pass-by-ref object without Far/Remotable`);
+      const desc = Object.entries(val).map(([k,v]) => `[${k}]=${v}`).concat(' , ');
+      console.log(`--- @@marshal: pass-by-ref [${desc}] object without Far/Remotable`, Error());
+      assert.fail(X`pass-by-ref object without Far/Remotable`);
       return REMOTE_STYLE;
     }
     case 'function': {
```

I've experimented with commenting out the asserts but leaving the logs in place, and then grepping the output. Let me know if I can help out, happy to pair on this.

### CODE REVIEW

**Comment by michaelfig** [comment]
I'll gladly peer with you tomorrow.  The only error I saw on trivial inspection of the test output was handled by you in a different PR (the `SwingSet/src/vats/network/` changes).

File: packages/cosmic-swingset/lib/ag-solo/vats/vat-http.js [hunk 1]
```diff
@@ -113,7 +114,7 @@ export function buildRootObject(vatPowers) {
         ...decentralObjects, // TODO: Remove; replaced by .agoric
         ...privateObjects, // TODO: Remove; replaced by .local
         ...handyObjects,
-        agoric: { ...decentralObjects },
+        agoric: { ...decentralObjects }, // TODO: maybe needs Data()???
```
Comment (michaelfig) [inline packages/cosmic-swingset/lib/ag-solo/vats/vat-http.js:117]: Yes, needs Data().

**Comment by warner** [comment]
@michaelfig this is ready for proper review now. The safety check to `store`/`weakstore` will land in #2585 after this lands.

**Review by michaelfig — APPROVED** [review]
LGTM111

**Pull request merged**